modal-go/sandbox.go

```diff
@@ -213,15 +207,16 @@ func (s *sandboxServiceImpl) Create(ctx context.Context, app *App, image *Image,
 		return nil, err
 	}
 
-	var envSecret *Secret
-	if len(params.Env) > 0 {
-		envSecret, err = s.client.Secrets.FromMap(ctx, params.Env, nil)
-		if err != nil {
-			return nil, err
-		}
+	mergedSecrets, err := mergeEnvIntoSecrets(ctx, s.client, &params.Env, &params.Secrets)
+	if err != nil {
+		return nil, err
 	}
 
-	req, err := buildSandboxCreateRequestProto(app.AppID, image.ImageID, *params, envSecret)
+	mergedParams := *params
+	mergedParams.Secrets = mergedSecrets
+	mergedParams.Env = nil
```

Comment:
Is setting Env to nil required? It looks like `buildSandboxCreateRequestProto` will ignore it.

In any case, can we have a short comment here? Someone new to the codebase will not really understand why we do this. (Same comment for the JS side)